agent/router/router.go

```diff
@@ -280,6 +281,18 @@ func (r *Router) maybeInitializeManager(area *areaInfo, dc string) *Manager {
 
 // addServer does the work of AddServer once the write lock is held.
 func (r *Router) addServer(areaID types.AreaID, area *areaInfo, s *metadata.Server) error {
+	if areaID == types.AreaLAN {
```

Comment:
As much as I want to add this prevention, there's nothing stopping someone from having a node named `blah.dc1` in `dc1` which would have a wan full name of `blah.dc1.dc1`.  At least 2 tests broke on this safety check.